tfjs-backend-webgpu/src/lrn_grad_webgpu.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {getMainHeaderString as main, WebGPUProgram} from './webgpu_program';
+import {computeDispatch, flatDispatchLayout} from './webgpu_util';
+
+export class LRNGradProgram implements WebGPUProgram {
+  outputShape: number[] = [];
+  shaderKey: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  variableNames = ['inputImage', 'outputImage', 'dy'];
+  uniforms =
+      'depth : i32, depthRadius : i32, bias : f32, alpha : f32, beta : f32,';
+  workgroupSize: [number, number, number] = [64, 1, 1];
+  size = true;
+
+  constructor(inputShape: number[]) {
+    this.outputShape = inputShape;
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    this.dispatch = computeDispatch(
+        this.dispatchLayout, this.outputShape, this.workgroupSize);
+    this.shaderKey = 'lrn_grad';
+  }
+
+  getUserCode(): string {
+    const userCode = `
+    ${main('index')} {
+      if (index < uniforms.size) {
+        let coords = getOutputCoords();
+        let b = coords[0];
+        let r = coords[1];
+        let c = coords[2];
+
+        var result = 0.0;
+        for (var d = 0; d < uniforms.depth; d++) {
+          let depthBegin = max(0, d - uniforms.depthRadius);
+          let depthEnd = min(uniforms.depth, d + uniforms.depthRadius + 1);
+
+          let MIN_DEPTH_BEGIN = 0;
+          let MAX_DEPTH_END = uniforms.depth;
+
+          var norm = 0.0;
+          for (var k = MIN_DEPTH_BEGIN; k < MAX_DEPTH_END; k++) {
+            if (k < depthBegin){
+              continue;
+            }
+            else if (k >= depthBegin && k < depthEnd) {
+              norm += getInputImage(b, r, c, k) * getInputImage(b, r, c, k);
+            }
+            else {
+              break;
+            }
+          }
+
+          norm = uniforms.alpha * norm + uniforms.bias;
+
+          for(var k = MIN_DEPTH_BEGIN; k < MAX_DEPTH_END; k++){
+            if (k < depthBegin){
+              continue;
+            }
+            else if (k >= depthBegin && k < depthEnd){
+              var dyi = -2.0 * uniforms.alpha * uniforms.beta
+                * getInputImage(b ,r ,c, k) * getOutputImage(b, r, c, d) / norm;
```

Comment:
Nit
```suggestion
                * getInputImage(b, r, c, k) * getOutputImage(b, r, c, d) / norm;
```
You may also fix the WebGL code.